utils/src/server/rpc_module.rs

```diff
@@ -69,41 +94,25 @@ impl Methods {
 	/// Merge two [`Methods`]'s by adding all [`SyncMethod`]s and [`AsyncMethod`]s from `other` into `self`.
 	/// Fails if any of the methods in `other` is present already.
 	pub fn merge(&mut self, other: Methods) -> Result<(), Error> {
-		for name in other.method_types.keys() {
+		for name in other.callbacks.keys() {
 			self.verify_method_name(name)?;
 		}
 
-		for (name, callback) in other.sync_methods {
-			self.sync_methods.insert(name, callback);
-			self.method_types.insert(name, MethodType::Sync);
-		}
-
-		for (name, callback) in other.async_methods {
-			self.async_methods.insert(name, callback);
-			self.method_types.insert(name, MethodType::Async);
+		for (name, callback) in other.callbacks {
+			self.callbacks.insert(name, callback);
 		}
 
 		Ok(())
 	}
 
-	/// Returns the type of the method handler, if any.
-	pub fn method_type(&self, method_name: &str) -> Option<&MethodType> {
-		self.method_types.get(method_name)
-	}
-
-	/// Returns the synchronous method.
-	pub fn sync_method(&self, method_name: &str) -> Option<&SyncMethod> {
-		self.sync_methods.get(method_name)
-	}
-
-	/// Returns the asynchronous method.
-	pub fn async_method(&self, method_name: &str) -> Option<&AsyncMethod> {
-		self.async_methods.get(method_name)
+	/// Returns the method callback.
+	pub fn method(&self, method_name: &str) -> Option<&MethodCallback> {
+		self.callbacks.get(method_name)
 	}
 
 	/// Returns a `Vec` with all the method names registered on this server.
-	pub fn method_names(&self) -> Vec<String> {
-		self.method_types.keys().map(|name| name.to_string()).collect()
+	pub fn method_names(&self) -> Vec<&'static str> {
+		self.callbacks.keys().map(|name| *name).collect()
```

Comment:
is this `deref` needed? does it work with `self.callbacks.keys().copied().collect()?

alternatively we can do do `impl IntoIterator<Item = &'static str>`

Reply:
Yeah, was thinking about that, or just `impl Iterator<Item = &'static str>`, I don't know what the lifetime requirements for this are in substrate.